luxonis_ml/data/exporters/exporter_utils.py

```diff
@@ -0,0 +1,252 @@
+import shutil
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import cv2
+import numpy as np
+import polars as pl
+from loguru import logger
+from pycocotools import mask
+
+if TYPE_CHECKING:
+    from luxonis_ml.data.datasets.luxonis_dataset import LuxonisDataset
+    from luxonis_ml.data.exporters.base_exporter import BaseExporter
+
+
+class PreparedLDF:
+    """Lightweight container for LDF data, ready for export."""
+
+    def __init__(
+        self,
+        splits: dict[str, Any],
+        processed_df: pl.DataFrame,
+        grouped_image_sources: pl.DataFrame,
+    ):
+        self.splits = splits
+        self.processed_df = processed_df
+        self.grouped_image_sources = grouped_image_sources
+
+    @classmethod
+    def from_dataset(cls, ldf: "LuxonisDataset") -> "PreparedLDF":
```

Comment:
It would be better to rename `ldf` to `dataset` to be consistent with the rest of the codebase.